web-client/src/app/state/session-xrpl.service.ts

```diff
@@ -77,6 +88,85 @@ export class SessionXrplService {
     return await this.sendTransaction(preparedTx);
   }
 
+  /**
+   * Sign and send a `TrustSet` transaction from the active session's wallet.
+   *
+   * @see https://xrpl.org/trustset.html
+   * @see XrplService.createUnsignedTrustSetTx
+   */
+  async sendTrustSetTx(
+    limitAmount: IssuedCurrencyAmount
+  ): Promise<xrpl.TxResponse> {
+    const { wallet } = this.sessionQuery.assumeActiveSession();
+
+    const preparedTx: xrpl.TrustSet = await withLoggedExchange(
+      'SessionXrplService.sendTrustSetTx: XrplService.createUnsignedTrustSetTx:',
+      async () =>
+        await this.xrplService.createUnsignedTrustSetTx(
+          wallet.xrpl_account.address_base58,
+          limitAmount
+        ),
+      { from: wallet.xrpl_account.address_base58, limitAmount }
+    );
+
+    return await this.sendTransaction(preparedTx);
+  }
+
+  /**
+   * Check trustline opt-in for each of this account's trust lines.
+   *
+   * @return The responses to `TrustSet` transactions sent out (empty if none sent)
+   * @see checkTrustlineOptIn
+   */
+  async checkTrustlineOptIns(): Promise<xrpl.TxResponse[]> {
+    // TODO(Pi): Check for necessary owner reserves before sending.
+    //           See: https://xrpl.org/reserves.html
+
+    const trustLines =
+      (await firstValueFrom(this.sessionQuery.xrplTrustlines)) ?? [];
+
+    const txResponses: xrpl.TxResponse[] = [];
+    for (const trustLine of trustLines) {
+      ifDefined(await this.checkTrustlineOptIn(trustLine), (txResponse) =>
+        txResponses.push(txResponse)
+      );
+    }
+    return txResponses;
+  }
+
+  /**
+   * Helper: Opt in to the given trust-line.
+   *
+   * This sends a `TrustSet` transaction matching the peer's limit
+   * if the active session's wallet's limit is zero.
+   *
+   * @return the `TrustSet` response, or undefined
+   */
+  async checkTrustlineOptIn(
```

Comment:
Name and docs don't match up

Reply:
Good catch!

Should be resolved by: [0f27d67..c497b29e](https://github.com/ntls-io/nautilus-wallet/compare/0f27d67..c497b29e)